tool/gravity/cli/operation.go

```diff
@@ -249,7 +260,49 @@ func (r *backendOperations) init(clusterBackend storage.Backend) error {
 		r.operations[op.ID] = (ops.SiteOperation)(op)
 	}
 	r.clusterOperation = (*ops.SiteOperation)(&clusterOperations[0])
-	r.operations[r.clusterOperation.ID] = *r.clusterOperation
+	return nil
+}
+
+func (r *backendOperations) listUpdateOperation(environ LocalEnvironmentFactory) error {
+	env, err := environ.NewUpdateEnv()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	defer env.Close()
+	r.getOperationAndUpdateCache(env.Backend, log.WithField("context", "update"))
+	return nil
+}
+
+func (r *backendOperations) listJoinOperation(environ LocalEnvironmentFactory) error {
+	env, err := environ.NewJoinEnv()
+	if err != nil && !trace.IsConnectionProblem(err) {
+		return trace.Wrap(err)
+	}
+	if env == nil {
+		// Do not fail for timeout errors.
+		// Timeout error means the directory is used by the active installer process
+		// which means, it's the installer environment, not joining node's
+		return nil
+	}
+	defer env.Close()
+	r.getOperationAndUpdateCache(env.Backend, log.WithField("context", "expand"))
+	return nil
+}
+
+func (r *backendOperations) listInstallOperation() error {
```

Comment:
Nit: This method doesn't return anything so we can do away with the return parameter.